website/addons/base/views.py

```diff
@@ -202,20 +191,20 @@ def get_auth(auth, **kwargs):
 
     try:
         credentials = provider_settings.serialize_waterbutler_credentials()
-        settings = provider_settings.serialize_waterbutler_settings()
+        waterbutler_settings = provider_settings.serialize_waterbutler_settings()
     except exceptions.AddonError:
         log_exception()
         raise HTTPError(httplib.BAD_REQUEST)
 
-    return {
+    return jwt.encode({
         'auth': make_auth(auth.user),  # A waterbutler auth dict not an Auth object
         'credentials': credentials,
-        'settings': settings,
+        'settings': waterbutler_settings,
         'callback_url': node.api_url_for(
             ('create_waterbutler_log' if not node.is_registration else 'registration_callbacks'),
             _absolute=True,
         ),
-    }
+    }, settings.WATERBUTLER_JWT_SECRET, algorithm=settings.WATERBUTLER_JWT_ALGORITHM)
```

Comment:
For added security set expiration to 30s (https://github.com/jpadilla/pyjwt#expiration-time-claim)